htdocs/takepos/pay.php

```diff
@@ -51,11 +53,60 @@
 	accessforbidden();
 }
 
+if (!empty($conf->stripe->enabled)) {
+	$service = 'StripeTest';
+	$servicestatus = 0;
+	if (!empty($conf->global->STRIPE_LIVE) && !GETPOST('forcesandbox', 'alpha')) {
+		$service = 'StripeLive';
+		$servicestatus = 1;
+	}
+
+	// Force to use the correct API key
+	global $stripearrayofkeysbyenv;
+	$site_account = $stripearrayofkeysbyenv[$servicestatus]['publishable_key'];
+
+	$stripe = new Stripe($db);
+	$stripeacc = $stripe->getStripeAccount($service); // Get Stripe OAuth connect account (no remote access to Stripe here)
+	$stripecu = $stripe->getStripeCustomerAccount($object->id, $servicestatus, $site_account); // Get remote Stripe customer 'cus_...' (no remote access to Stripe here)
+	$keyforstripeterminalbank = "CASHDESK_ID_BANKACCOUNT_STRIPETERMINAL".$_SESSION["takeposterminal"];
+	?>
+<script src="https://js.stripe.com/terminal/v1/"></script>
+<script>
+var terminal = StripeTerminal.create({
+  onFetchConnectionToken: fetchConnectionToken,
+  onUnexpectedReaderDisconnect: unexpectedDisconnect,
+});
+function unexpectedDisconnect() {
+  // In this function, your app should notify the user that the reader disconnected.
+  // You can also include a way to attempt to reconnect to a reader.
+  console.log("Disconnected from reader")
+}
+function fetchConnectionToken() {
+	<?php
+	$urlconnexiontoken = DOL_URL_ROOT.'/stripe/ajax/ajax.php?action=getConnexionToken&servicestatus='.$servicestatus;
+	if (!empty($conf->global->STRIPE_LOCATION)) $urlconnexiontoken .= '&location='.$conf->global->STRIPE_LOCATION;
+	if (!empty($stripeacc)) $urlconnexiontoken .= '&stripeacc='.$stripeacc;
+	?>
+  // Do not cache or hardcode the ConnectionToken. The SDK manages the ConnectionToken's lifecycle.
+  return fetch('<?php echo $urlconnexiontoken; ?>', { method: "POST" })
+	.then(function(response) {
+	  return response.json();
+	})
+	.then(function(data) {
+	  return data.secret;
+	});
+}
+</script>
+<?php }
 
 /*
  * View
  */
 
+if (!empty($conf->stripe->enabled) && (empty($conf->global->STRIPE_LIVE) || GETPOST('forcesandbox', 'alpha'))) {
```

Comment:
If "CASHDESK_ID_BANKACCOUNT_STRIPETERMINAL".$terminaltouse  is not set, it means we didn't configure to use stripe for payment on takepos. Right ?
If right, can you add a condition here to avoid warning when not required.

Reply:
@ptibogxiv Can you check this ?